src/bb/image-io/rt_u3v.h

```diff
@@ -322,443 +300,546 @@ class U3V {
+            devices_[1].camera_ = fake_camera1;
+        }
+        // Config fake cameras
+        for (int i = 0;i< num_sensor_;i++){
+            // setting the params if it is not zero
+            log::info("Width {}, Height {} PixelFormat {}...", width, height, pixel_format_);
+            arv_device_set_integer_feature_value (devices_[i].device_, "Width", width, &err_);
+            arv_device_set_integer_feature_value (devices_[i].device_, "Height", height, &err_);
+            arv_device_set_float_feature_value (devices_[i].device_, "AcquisitionFrameRate",fps, &err_);
+            if (pixel_format_ != "Mono8")
+                arv_device_set_string_feature_value(devices_[i].device_, "PixelFormat", pixel_format.c_str(), &err_);
+            devices_[i].u3v_payload_size_ =  arv_device_get_integer_feature_value (devices_[i].device_, "PayloadSize", &err_);
+            auto px =arv_device_get_integer_feature_value(devices_[i].device_, "PixelFormat", &err_);
+            auto fps = arv_device_get_float_feature_value(devices_[i].device_, "AcquisitionFrameRate", &err_);
+            struct rawHeader header=  { 1, width, height,
+                1, 1, 1, 1, 1, 1, 0, 0, 0, 0,
+                width, height, width, height, static_cast<float>(fps), px};
+            devices_[i].header_info_ = header;
+            devices_[i].image_payload_size_ = devices_[i].u3v_payload_size_;
+            devices_[i].frame_count_  = 0;
+
+        }
+
+        // Start streaming and start acquisition
+        devices_[0].stream_ = arv_device_create_stream (devices_[0].device_, NULL, NULL, &err_);
+        if (num_sensor_==2){
+            devices_[1].stream_ = arv_device_create_stream (devices_[1].device_, NULL, NULL, &err_);
+        }
+
+        for (auto i=0; i<devices_.size(); ++i) {
+            arv_device_execute_command(devices_[i].device_, "AcquisitionStart", &err_);
+            log::info("\tFake Device {}::{} : {}", i, "Command", "AcquisitionStart");
+        }
+    };
+
+};
+
+class U3VRealCam: public U3V{
+public:
+    static U3V & get_instance(const std::string& id,
+                              int32_t num_sensor,
+                              bool frame_sync,
+                              bool realtime_display_mode,
+                              bool sim_mode = false,
+                              int32_t width = 640,
+                              int32_t height = 480,
+                              float_t fps = 25.0,
+                              const std::string& pixel_format = "Mono8"
+                              )
+    {
+        if (instances_.count(id) == 0) {
+            ion::log::info("Create U3VRealCam instance: {}", id);
+            instances_[id] = std::unique_ptr<U3V>(new U3VRealCam(num_sensor, frame_sync, realtime_display_mode, sim_mode, width, height, fps, pixel_format));
+        }
+
+        return *instances_[id].get();
+    }
+
+
+    void set_realtime(std::vector<ArvBuffer *> &bufs, int timeout_us = 3 * 1000 * 1000){
```

Comment:
This method doesn't look like "setting realtime". Maybe `consume_old_buffer` or `ensure_newest_buffer` or something else similar is better?